src/api/modules/dispatch/dispatch.background.ts

```diff
@@ -112,8 +112,21 @@ const background: BackgroundModuleFunction<ReturnType> = async (
     };
   } catch (err) {
     if (isError(err) && err.message === ERR_MSG_USER_CANCELLED_AUTH) {
+      freeDecryptedWallet(keyfile);
+
       throw err;
     }
+  }
+
+  try {
+    // grab the user's keyfile again:
+    decryptedWallet = await getActiveKeyfile(appData);
```

Comment:
With the previous implementation with a single `try-catch`, if the first upload attempt fails, the second one would fail too, because `ensureAllowanceDispatch()` internally has another `try-catch` that calls `freeDecryptedWallet()`, so in the second upload attempt there would be no keyfile. So we need to get the keyfile again in the second attempt.